### PR TITLE
[WIP] Added examples for creating the secrets

### PR DESCRIPTION
**Pull request by padiazg**
Add example on how to create the secrets needed for using authentication on the API Gateway

Signed-off-by: Patricio Diaz padiazg@gmail.com

### CODE REVIEW

File: docs/reference/authentication.md [hunk 1]
```diff
@@ -14,6 +14,22 @@ We recommend using basic authentication and a strong password to protect the `/s
...
+
+Option 1 (simpler, but doesn't works under Windows and docker complains about security)
+```bash
+echo myusername | docker secret create basic-auth-user -
+echo secr3t | docker secret create basic-auth-password -
+```
+
+Option 2 (safer, must work under any OS)
+```bash
+echo myusername > myusername.txt
```
Comment (rgee0) [inline docs/reference/authentication.md]: Prepend lines that show commands with `$ `
merge the two commands as we do elsewhere 
```
command1 && \
   command2
```

File: docs/reference/authentication.md [hunk 2]
```diff
@@ -14,6 +14,22 @@ We recommend using basic authentication and a strong password to protect the `/s
 
 The API Gateway as of version 0.8.2 provides built-in basic authentication. To use it set the environmental variable `basic_auth` to true. Then create two secrets named `basic-auth-user` and `basic-auth-password`.
 
+**To create the secrets there are two options:**
+
+Option 1 (simpler, but doesn't works under Windows and docker complains about security)
```
Comment (rgee0) [inline docs/reference/authentication.md]: Typo in `works`
`docker` should be `Docker`

File: docs/reference/authentication.md [hunk 3]
```diff
@@ -14,6 +14,22 @@ We recommend using basic authentication and a strong password to protect the `/s
...
+```bash
+echo myusername | docker secret create basic-auth-user -
```
Comment (rgee0) [inline docs/reference/authentication.md]: Prepend lines that show commands with $
Do you find that echo appends a `\n`?  Is this desirable?

**Comment by rgee0** [comment]
Not sure what happened but Github lost half my review comments, so I've added them as inline comments.

**Comment by alexellis** [comment]
@padiazg @rgee0 what do you mean "in two places".. what is the second place we show how to create the basic auth secrets?

We need to have something showing how to set up basic auth ASAP in the docs as it seems like there's a few people who need help with this level of detail.

**Comment by rgee0** [comment]
Are we intending to explicitly describe the commands required to create a secret wherever a secret name varies?

_Then create two secrets named `basic-auth-user` and `basic-auth-password` using the appropriate command from the [secrets](http://docs.openfaas.com/reference/secrets/) section._

File: docs/reference/authentication.md (hunk 1 above)
Comment (padiazg) [reply docs/reference/authentication.md]: Thanks for pointing these suggestions, this is my fist contibution. Will preppend the commands with ```$```, but prefer to leave each command separated beacuse ```&&`` works in Linux/Bash but not in Windows/Cmd.

**Comment by rgee0** [comment]
I still have a question around whether explicitly describing secret creation in every situation that the secret name varies is the right approach.

Also, basic auth is enabled by default which may negate the need for this section.

**Comment by padiazg** [comment]
Maybe you are right, maybe this section down't has a reason to exist, but please understand that  not everyone has your skill, even I don´t have your skill. Came to this section once I needed it, and the info didn´t helped me because I use different OS to try, develop, test and deploy to production. I thought that this would be a good opportunity to share something I know. 
I think it's not so bad to have this duplicated as you point it, since it will serve to fix the concept of creating secrets. The section on secrets is very useful, if you know the concept perfectly, but if you are a novice or never used them, it can be a source of disappointment  if you can not apply the instructions, an example is useful here.

**Comment by alexellis** [comment]
Hi @padiazg I think we've covered this in the deploy_stack.sh file now with automation. Can this PR be closed?

**Comment by padiazg** [comment]
Sure. I think this PR has no purpose right now.

**Pull request closed without merging**